### PR TITLE
Ensure end time is after start time

### PR DESCRIPTION
**Pull request by yifeiwu**
Ensure end time is after start time for an event.

### CODE REVIEW

**Review by xuebingli — APPROVED** [review]
👍 assuming that time only flows in one direction and that very direction is forward :trollface:

**Review by xuebingli — APPROVED** [review]
👍 assuming that time only flows in one direction and that very direction is forward :trollface:

**Comment by yifeiwu** [comment]
Go home Dr Strange, you're drunk

**Review by nchieng — APPROVED** [review]

**Review by ragurney — APPROVED** [review]

**Review by craig-day — CHANGES_REQUESTED** [review]
please add a server side validation too

**Comment by yifeiwu** [comment]
@craig-day added server side tests

**Comment by xuebingli** [comment]
@craig-day 💇‍♂️

File: app/models/event.rb [hunk 1]
```diff
@@ -44,9 +45,11 @@ def remove_user!(user)
   private
 
   def max_capacity
-    if capacity && users.size > capacity
-      errors.add(:users, "total can't be above capacity")
-    end
+    errors.add(:users, "total can't be above capacity") if capacity && users.size > capacity
```
Comment (craig-day) [inline app/models/event.rb]: don't love this style change, the line is pretty long now.

File: app/models/event.rb [hunk 2]
```diff
@@ -44,9 +45,11 @@ def remove_user!(user)
...
+  end
+
+  def time_sequentiality
+    errors.add(:ends_at, "should not end before it starts") if starts_at && ends_at && ends_at <= starts_at
```
Comment (craig-day) [inline app/models/event.rb]: I would prefer an if block, not a trailing if when its this long.

**Review by craig-day — APPROVED** [review]

**Pull request merged**